setup.js

```diff
@@ -0,0 +1,123 @@
+var fs = require('fs');
+var prompt = require('prompt');
+var template = require('lodash.template');
+var merge = require('lodash.merge');
+var rmraf = require('rimraf');
+
+var argv = process.argv;
+
+var defaults = {
+  version: '0.0.1',
+  yourName: 'We Are Fractal',
+  yourEmail: 'contact@wearefractal.com',
+  yourDomain: 'http://wearefractal.com',
+  gitUserName: 'wearefractal'
+};
+
+
+var questions = {
+  properties: {
+    projectName: {
+      pattern: /^[a-zA-Z\-]+$/,
+      message: 'name must not contain spaces',
+      required: true
+    },
+    description: {},
+    version: {
+      default: defaults.version
+    },
+    yourName: {
+      required: true,
+      default: defaults.yourName
+    },
+    yourEmail: {
+      required: true,
+      default: defaults.yourEmail
+    },
+    yourDomain: {
+      required: true,
+      default: defaults.yourDomain
+    },
+    gitUserName: {
+      required: true,
+      default: defaults.gitUserName
+    }
+  }
+};
+
+prompt.start();
+
+if (argv[2] == '--use-defaults') {
+  console.log('Using defaults \n');
+
+  var defaultQuestions = {
+    properties: {
+      projectName: {
+        pattern: /^[a-zA-Z\-]+$/,
+        message: 'name must not contain spaces',
+        required: true
+      },
+      description: {},
+    }
+  };
+
+  return prompt.get(defaultQuestions, function(err, data) {
+    data = merge(data, defaults);
+    save(data);
+  });
+}
+
+prompt.get(questions, function(err, data) {
+  save(data);
+});
+
+
+function save(data) {
+  if (data === null || data === undefined) {
+    return console.log('\n Error: inputs are undefined');
+  }
+  data.year = new Date().getFullYear();
+  var files = [
+    './package.json.bk',
+    './README.md.bk',
+    './LICENSE',
+    './test/index.js'
+  ];
+
+  files.forEach(function(v, k, a) {
+
+    if (v.indexOf('.bk') > -1) {
+      var newName = v.replace('.bk', '');
+      return fs.rename(v, newName, function(err) {
```

Comment:
everything is already synchronous, not really any reason to use an async rename here. renameSync would work. since all templated files will end with .bk this check becomes redundant